Rest/Controller/ContentController.php

```diff
@@ -0,0 +1,199 @@
+    protected $configResolver;
+
+    /** @var ImageVariationService */
+    protected $imageVariationService;
+
+    /** @var UrlGenerator */
+    protected $generator;
+
+    /**
+     * @param ConfigResolverInterface $configResolver
+     * @param ImageVariationService   $imageVariationService
+     * @param UrlGenerator            $generator
+     */
+    public function __construct(ConfigResolverInterface $configResolver, ImageVariationService $imageVariationService, UrlGenerator $generator)
+    {
+        $this->configResolver = $configResolver;
+        $this->imageVariationService = $imageVariationService;
+        $this->generator = $generator;
+    }
+
+    /**
+     * Prepares content for ContentDataValue class.
+     *
+     * @param $contentId
+     *
+     * @return ContentDataValue
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException     if the content, version with the given id and languages or content type does not exist
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the user has no access to read content and in case of un-published content: read versions
+     */
+    public function getContent($contentId)
+    {
+        $contentIds = explode(',', $contentId);
+        $content = $this->prepareContent($contentIds);
+
+        return new ContentDataValue($content);
+    }
+
+    /**
+     * Prepare nested content array.
+     *
+     * @param $contentIds
+     * @param int $nesting Maximum nesting level for container content.
+     *
+     * @return array
+     */
+    protected function prepareContent($contentIds)
+    {
+        $contentService = $this->repository->getContentService();
+        $locationService = $this->repository->getLocationService();
+        $contentTypeService = $this->repository->getContentTypeService();
+
+        $requestLanguage = $this->request->get('lang');
+        $requestedFields = $this->request->get('fields');
+
+        $content = [];
+
+        foreach ($contentIds as $contentId) {
+            $contentValue = $contentService->loadContent((int) $contentId, [$requestLanguage]);
+            $contentType = $contentTypeService->loadContentType($contentValue->contentInfo->contentTypeId);
+            $location = $locationService->loadLocation($contentValue->contentInfo->mainLocationId);
+            $user = $this->repository->getUserService()->loadUser($contentValue->contentInfo->ownerId);
+            $language = (is_null($requestLanguage)) ? $contentType->mainLanguageCode : $requestLanguage;
+
+            $content[$contentId] = [
+                'contentId' => (int) $contentId,
+                'contentTypeId' => $contentType->id,
+                'identifier' => $contentType->identifier,
+                'language' => $language,
+                'publishedDate' => $contentValue->contentInfo->publishedDate->format('c'),
+                'author' => $user->getFieldValue('first_name').' '.$user->getFieldValue('lasst_name'),
```

Comment:
As far as I remember `author` is a magic word required by YC? Am I right?
Besides you got typo here `lasst_name` ;)